lib/widgets/topic_list.dart

```diff
@@ -0,0 +1,301 @@
+      showErrorDialog(
+        context: context,
+        title: zulipLocalizations.errorFetchingTopics,
+        message: e.toString());
+    }
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in the models.
+      // This method was called because they just changed.
+    });
+  }
+
+  @override
+  void dispose() {
+    _unreadsModel?.removeListener(_modelChanged);
+    _model?.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_model == null || _model!.isLoading) {
+      return const Center(child: CircularProgressIndicator());
+    }
+
+    if (_model!.topics == null || _model!.topics!.isEmpty) {
+      return Center(
+        child: Text(ZulipLocalizations.of(context).noTopicsInChannel,
+          style: TextStyle(
+            color: DesignVariables.of(context).labelMenuButton,
+            fontSize: 16,
+          )),
+      );
+    }
+
+    final sortedTopics = List<GetStreamTopicsEntry>.from(_model!.topics!);
+    sortedTopics.sort((a, b) => b.maxId.compareTo(a.maxId));
+
+    return SafeArea(
+      child: ListView.builder(
+        itemCount: sortedTopics.length,
+        itemBuilder: (context, index) {
+          final topic = sortedTopics[index];
+          return TopicListItem(
+            streamId: widget.streamId,
+            topic: topic,
+          );
+        },
+      ),
+    );
+  }
+}
+
+class TopicListItem extends StatelessWidget {
+  const TopicListItem({
+    super.key,
+    required this.streamId,
+    required this.topic,
+  });
+
+  final int streamId;
+  final GetStreamTopicsEntry topic;
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final designVariables = DesignVariables.of(context);
+
+    final unreads = store.unreads.countInNarrow(
+      TopicNarrow(streamId, topic.name)
+    );
+    final hasUnreads = unreads > 0;
+
+    return Material(
+      type: MaterialType.transparency,
+      child: InkWell(
+        onTap: () {
+          Navigator.push(context, MessageListPage.buildRoute(
+            context: context,
+            narrow: TopicNarrow(streamId, topic.name),
+          ));
+        },
+        onLongPress: () {
+          showTopicActionSheet(
+            context,
+            channelId: streamId,
+            topic: topic.name,
+            someMessageIdInTopic: null,
+          );
+        },
+        child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
+          const SizedBox(width: 28),
+          Padding(
+            padding: const EdgeInsets.only(top: 10.0),
+            child: SizedBox(
+              height: 16,
+              width: 16,
+              child: topic.name.displayName.startsWith('✔')
+                ? Icon(
+                    ZulipIcons.check,
+                    size: 12,
+                    color: designVariables.textMessage,
+                  )
```

Comment:
The opacity of the icon does not match the design:

![image](https://github.com/user-attachments/assets/6116fe7a-2a69-4aa0-9afb-ba1d35c85f54)

Reply:
The size of the icon should be 16px.